crates/rune/src/compile/options.rs

```diff
@@ -61,8 +77,24 @@ impl Options {
             Some("v2") => {
                 self.v2 = it.next() != Some("false");
             }
+            Some("dynamic_fields") => {
+                let result = it.next() != Some("false");
+                if result {
+                    #[cfg(feature = "dynamic_fields")]
+                    {
+                        self.dynamic_fields = result;
+                    }
+                    #[cfg(not(feature = "dynamic_fields"))]
+                    {
+                        return Err(ParseOptionsErrors::DisabledFeatureOptionError {
+                            option: option.into(),
+                            required_feature: "dynamic_fields",
+                        });
+                    }
+                }
```

Comment:
`if cfg!(feature = "dynamic_fields") { /*  */ } else { /*  */ }` is probably cleaner here.

Reply:
> `if cfg!(feature = "dynamic_fields") { /* */ } else { /* */ }` is probably cleaner here.

Yeah easily sorted; it is just like this due to `ObjectIndexDynamicGet` originally being feature gated so it wouldn't compile with that if statement lol (kind of wish rust had a macro which mimicked if else cfg macro but actually ignored the invalid branch since its more sane to read.